### DISCUSSION TITLE
Is the TimescaleDB worker supposed to always be running?

### DISCUSSION THREAD

**Discussion by vickkhera** [supabase/supabase#14536] [Questions] [answered]
I was poking around what's running in my local DB by inspecting `pg_stat_activity`. I noticed that a _TimescaleDB Background Worker Launcher_ process is running. I have never enabled timescaledb extension and it is definitely showing as not enabled on the local dashboard nor the psql client with `\dx`.

Does just having the extension available suck up a backend worker slot? It is running as user `postgres` not associated with any database. Is there some other DB defined on my Pg cluster I cannot see that enables it internally (template1 does not have the extension installed either)?

```
postgres=> \dx
                                            List of installed extensions
        Name        | Version |   Schema   |                              Description
--------------------+---------+------------+------------------------------------------------------------------------
 pg_graphql         | 1.2.0   | graphql    | pg_graphql: GraphQL support
 pg_net             | 0.7.1   | extensions | Async HTTP
 pg_stat_statements | 1.10    | extensions | track planning and execution statistics of all SQL statements executed
 pgcrypto           | 1.3     | extensions | cryptographic functions
 pgjwt              | 0.2.0   | extensions | JSON Web Token API for Postgresql
 pgsodium           | 3.1.6   | pgsodium   | Pgsodium is a modern cryptography library for Postgres.
 plpgsql            | 1.0     | pg_catalog | PL/pgSQL procedural language
 supabase_vault     | 0.2.8   | vault      | Supabase Vault Extension
 uuid-ossp          | 1.1     | extensions | generate universally unique identifiers (UUIDs)
(9 rows)

postgres=> select * from pg_stat_activity where backend_type ilike '%timescale%';
-[ RECORD 1 ]----+---------------------------------------
datid            |
datname          |
pid              | 41
leader_pid       |
usesysid         | 10
usename          | postgres
application_name | TimescaleDB Background Worker Launcher
client_addr      |
client_hostname  |
client_port      |
backend_start    | 2023-05-22 14:30:35.594549+00
xact_start       |
query_start      |
state_change     |
wait_event_type  | Extension
wait_event       | Extension
state            |
backend_xid      |
backend_xmin     |
query_id         |
query            |
backend_type     | TimescaleDB Background Worker Launcher
```

**Comment by GaryAustin1** [comment]
I can confirm it is running on a new instance.  There seems to be code in supabase/postgres to test the extensions.  Total speculation, but I wonder if they start them, do some test, then turn them off.  Maybe that leaves behind a task.

But anyway your case is not unique.

**Comment by steve-chavez** [comment, marked as answer]
Yes, extensions that use background workers(running process) like Timescale must always be included in [shared_preloard_libraries](https://postgresqlco.nf/doc/en/param/shared_preload_libraries/).

> is definitely showing as not enabled on the local dashboard

That sounds like the actual fix for now. The dashboard should show it as enabled.

**Reply by vickkhera** [reply]
Thanks for the clarification.

It seems like it is "half" there -- the worker is running, but you cannot access the other functions of it. What would it mean to turn if off on the UI if it defaults to on? Seems like it (and pg_net, as it has a worker) ought not to be able to get disabled.

**Reply by steve-chavez** [reply]
Yeah, that's a current pg limitation. So to really "turn off" the bg worker you have to kill it - `pg_terminate_backend`. The drawback being that once you do that, you have to restart the database for it to live again.
